### ISSUE TITLE
No fixers have been defined. Try :ALEFixSuggest

### ISSUE THREAD

**Issue by thepreacher** [w0rp/ale#1878]
## Information

**VIM version**

NVIM v0.3.1
Operating System: macOS 10.13
### :ALEInfo

 Current Filetype: python
Available Linters: ['flake8', 'mypy', 'prospector', 'pycodestyle', 'pyflakes', 'pylint', 'pyls', 'pyre', 'vulture']
  Enabled Linters: ['flake8', 'mypy', 'pylint']
 Suggested Fixers: 
  'add_blank_lines_for_python_control_statements' - Add blank lines before control statements.
  'autopep8' - Fix PEP8 issues with autopep8.
  'black' - Fix PEP8 issues with black.
  'isort' - Sort Python imports with isort.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'yapf' - Fix Python files with yapf.
 Linter Variables:

let g:ale_python_flake8_change_directory = 1
let g:ale_python_flake8_executable = 'flake8'
let g:ale_python_flake8_options = ''
let g:ale_python_flake8_use_global = 0
let g:ale_python_mypy_executable = 'mypy'
let g:ale_python_mypy_ignore_invalid_syntax = 0
let g:ale_python_mypy_options = ''
let g:ale_python_mypy_use_global = 0
let g:ale_python_pylint_change_directory = 1
let g:ale_python_pylint_executable = 'pylint'
let g:ale_python_pylint_options = ''
let g:ale_python_pylint_use_global = 0
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%c

## What went wrong

I have the following ale and python config in my vimrc 
```
" Keep the sign gutter open at all times
let g:ale_sign_column_always = 1
" Set this variable to 1 to fix files when you save them.
let g:ale_fix_on_save = 1
```
```
" Check Python files with flake8 and pylint.
let g:ale_linters = {'python': ['mypy', 'flake8', 'pylint']}
" Fix Python files with black and isort.
let g:ale_fixers = {'python': ['black', 'isort']}
```
However although linting works, fixing the code as per `black` and `isort` as specified doesn't.

From the vim prompt when I try :ALEFix, I get the message 

> No fixers have been defined. Try :ALEFixSuggest

But as you can see I've specified ale_fixers for python and I also have the packages installed in my virtualenv environment. I am able to use them from the terminal but not from vim.

**Comment by w0rp** [comment]
What's the value of `b:ale_fixers`? What other plugins do you have installed? Make sure to include the entire output of `:ALEInfo` with issues.

**Comment by thepreacher** [comment]
Thanks for your responds @w0rp. What I did in the mean time was to reorganize my vimrc by moving language specific plugin configurations to their respective files in ftplugin. In doing so I created a python.vim file and moved the Ale configs for python into it. As a result I changed 

```
" Check Python files with flake8 and pylint.
let g:ale_linters = {'python': ['mypy', 'flake8', 'pylint']}
" Fix Python files with black and isort.
let g:ale_fixers = {'python': ['black', 'isort']}
```
to

```
" Check Python files with flake8 and pylint.
let b:ale_linters = {'python': ['mypy', 'flake8', 'pylint']}
" Fix Python files with black and isort.
let b:ale_fixers = {'python': ['black', 'isort']}
```
and that seems to have fixed it. After I got it working I deliberately restored the old vimrc and the problem surfaced again.

Infact I had this issues for all my languages elm, elixir etc but creating files for each in ftplugin has done the trick. 

So I'll leave this for a bit a close it later if there is no further development.
